File: 프로그래머스/unrated/181893. 배열 조각하기/배열 조각하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[] query) {
        int num = 0;
        for(int i : query) {
            if(num % 2 == 0) {
                arr = even(arr, i);
            }
            else {
                arr = odd(arr, i);
            }
            num++;
        }
        
        return arr;
    }

    public static int[] even(int[] arr, int query_value) {
        int[] sub = new int[query_value + 1];
        for(int i = 0; i <= query_value; i++) {
            sub[i] = arr[i];
        }
        return sub;
    }

    public static int[] odd(int[] arr, int query_value) {
        int[] sub = new int[arr.length - query_value];
        for(int i = query_value; i < arr.length; i++) {
            sub[i - query_value] = arr[i];
        }
        return sub;
    }
}

File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.py
# 정수 배열 arr, query
def solution(arr, query):
    answer = []
    for i in range(len(query)):
        if i % 2 == 0:
            # query[i] 뒷부분 삭제
            arr = arr[:query[i]+1]
        else:
            # query[i] 앞부분 삭제
            arr = arr[query[i]:]
    return arr